packages/react-aria-components/test/ListBox.test.js

```diff
@@ -1321,11 +1324,225 @@ describe('ListBox', () => {
+    });
+
+    describe('virtualized', () => {
+      let items = [];
+      for (let i = 0; i < 50; i++) {
+        items.push({name: 'Foo' + i});
+      }
+      let clientWidth, clientHeight;
+
+      beforeAll(() => {
+        clientWidth = jest.spyOn(window.HTMLElement.prototype, 'clientWidth', 'get').mockImplementation(() => 100);
+        clientHeight = jest.spyOn(window.HTMLElement.prototype, 'clientHeight', 'get').mockImplementation(() => 100);
+      });
+
+      afterEach(() => {
+        act(() => {jest.runAllTimers();});
+      });
+
+      afterAll(() => {
+        clientWidth.mockReset();
+        clientHeight.mockReset();
+      });
+
+      function VirtualizedAsyncListbox(props) {
+        let {items, isLoading, onLoadMore, ...listBoxProps} = props;
+        return (
+          <Virtualizer
+            layout={ListLayout}
+            layoutOptions={{
+              rowHeight: 25,
+              loaderHeight: 30
+            }}>
+            <ListBox
+              {...listBoxProps}
+              aria-label="async virtualized listbox"
+              renderEmptyState={() => renderEmptyState()}>
+              <Collection items={items}>
+                {(item) => (
+                  <ListBoxItem id={item.name}>{item.name}</ListBoxItem>
+                )}
+              </Collection>
+              <UNSTABLE_ListBoxLoadingSentinel isLoading={isLoading} onLoadMore={onLoadMore}>
+                Loading...
+              </UNSTABLE_ListBoxLoadingSentinel>
+            </ListBox>
+          </Virtualizer>
+        );
+      };
+
+      it('should always render the sentinel even when virtualized', () => {
+        let tree = render(<VirtualizedAsyncListbox isLoading items={items} />);
+        let listboxTester = testUtilUser.createTester('ListBox', {root: tree.getByRole('listbox')});
+        let options = listboxTester.options();
+        expect(options).toHaveLength(8);
+        let loaderRow = options[7];
+        expect(loaderRow).toHaveTextContent('Loading...');
+        expect(loaderRow).toHaveAttribute('aria-posinset', '51');
+        expect(loaderRow).toHaveAttribute('aria-setSize', '51');
+        let loaderParentStyles = loaderRow.parentElement.style;
+
+        // 50 items * 25px = 1250
+        expect(loaderParentStyles.top).toBe('1250px');
+        expect(loaderParentStyles.height).toBe('30px');
+
+        let sentinel = within(loaderRow.parentElement).getByTestId('loadMoreSentinel');
+        expect(sentinel.parentElement).toHaveAttribute('inert');
+      });
+
+      // TODO: for some reason this tree renders empty if ran with the above test...
+      // Even if the above test doesn't do anything within it, the below tree won't render with content until the above test
+      // is fully commented out (aka even the it(...))
+      // It thinks that the contextSize is 0 and never updates
+      it.skip('should not reserve room for the loader if isLoading is false', () => {
```

Comment:
Not quite sure why this is happening... As stated, the below test only renders the listbox items if the above test is completely commented out. Even if the above test doesn't have any contents (aka just an empty `it()`) this test still fails to render the ListBox content